File: Languages/R/Day013.r
# Languages - R - Day 13
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a function that takes an array of numbers, calculates the sum of all elements, and returns the result. Call this function with a sample array and print the result.

# Define a function that calculates the sum of an array
sum_of_elements <- function(arr) {
  sum(arr)
}

# Prompt the user for input
input_numbers <- readline(prompt = "Enter a list of numbers separated by commas (e.g., 1,2,3): ")

# Validate the input
if (nchar(input_numbers) == 0 || !grepl("^\\s*[0-9]+(\\s*,\\s*[0-9]+)*\\s*$", input_numbers)) {
  cat("Invalid input. Please enter a list of numbers separated by commas.\n")
  quit(save = "no", status = 1)
}

# Convert the input string into a numeric array
numbers <- as.numeric(unlist(strsplit(gsub("\\s+", "", input_numbers), ",")))

# Check if any elements failed to convert
if (any(is.na(numbers))) {
  cat("Invalid input. Ensure all elements are valid numbers.\n")
  quit(save = "no", status = 1)
}

# Calculate and display the sum
result <- sum_of_elements(numbers)
cat("The sum of all elements is:", result, "\n")

File: Languages/Swift/Day013.swift
// Languages - Swift - Day 13
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that takes an array of numbers, calculates the sum of all elements, and returns the result. Call this function with a sample array and print the result.

// Define the function
func sumOfArray(_ numbers: [Int]) -> Int {
    var sum = 0
    for number in numbers {
        sum += number
    }
    return sum
}

// Inform the user about the input format
print("Enter numbers separated by spaces (e.g., 1 2 3):")

// Read user input
if let input = readLine() {
    // Convert input to an array of integers
    let numbers = input.components(separatedBy: " ").compactMap { Int($0) }
    
    // Validate the input
    if numbers.isEmpty {
        print("Invalid input. Please enter a valid list of integers separated by spaces.")
    } else {
        // Call the function and print the result
        let result = sumOfArray(numbers)
        print("The sum of the array is: \(result)")
    }
} else {
    print("No input received. Please try again.")
}
